projects/packages/backup/tests/php/test-rest-controller.php

```diff
@@ -0,0 +1,246 @@
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+
+namespace Automattic\Jetpack\Backup;
+
+use Automattic\Jetpack\Connection\Rest_Authentication as Connection_Rest_Authentication;
+use PHPUnit\Framework\TestCase;
+use WorDBless\Options as WorDBless_Options;
+use WP_REST_Request;
+use WP_REST_Server;
+
+/**
+ * Unit tests for the REST_Controller class.
+ *
+ * @package automattic/jetpack-backup
+ */
+class Test_REST_Controller extends TestCase {
+
+	/**
+	 * REST Server object.
+	 *
+	 * @var WP_REST_Server
+	 */
+	private $server;
+
+	/**
+	 * Setting up the test.
+	 *
+	 * @before
+	 */
+	public function set_up() {
+		global $wp_rest_server;
+
+		$wp_rest_server = new WP_REST_Server();
+		$this->server   = $wp_rest_server;
+
+		// Register REST routes.
+		add_action( 'rest_api_init', array( 'Automattic\\Jetpack\\Backup\\REST_Controller', 'register_rest_routes' ) );
```

Comment:
I had to add this here because `actions.php` is auto-loaded by composer and when running unit tests at the time it gets loaded, WP is not available yet (exiting early because `ABSPATH` is not defined).

Now I wonder if this is ok in general or the way `actions.php` is loaded could cause any issues outside of unit testing.

cc @kbrown9  Just to confirm!

Reply:
I can't think of any issues that would be caused by the way `actions.php` is loaded.

Are you concerned that the same issue that you're facing with unit testing could also happen outside of unit testing? I don't think that's possible. A plugin will load the autoloader, which will load the `actions.php` file. The plugin itself will be loaded by WP. I can't think of a situation where the `actions.php` file would get loaded before WP.

Reply:
That was my understanding as well, just wanted to make sure I'm not missing anything! Many thanks for confirming, Kim!